src/agentscope/models/zhipu_model.py

```diff
@@ -0,0 +1,348 @@
+    def __call__(
+        self,
+        texts: str,
+        **kwargs: Any,
+    ) -> ModelResponse:
+        """Embed the messages with ZhipuAI embedding API.
+
+        Args:
+            texts (`str`):
+                The messages used to embed.
+            **kwargs (`Any`):
+                The keyword arguments to ZhipuAI embedding API,
+                e.g. `encoding_format`, `user`. Please refer to
+                https://open.bigmodel.cn/dev/api#text_embedding
+                for more detailed arguments.
+
+        Returns:
+            `ModelResponse`:
+                A list of embeddings in embedding field and the
+                raw response in raw field.
+
+        Note:
+            `parse_func`, `fault_handler` and `max_retries` are reserved for
+            `_response_parse_decorator` to parse and check the response
+            generated by model wrapper. Their usages are listed as follows:
+                - `parse_func` is a callable function used to parse and check
+                the response generated by the model, which takes the response
+                as input.
+                - `max_retries` is the maximum number of retries when the
+                `parse_func` raise an exception.
+                - `fault_handler` is a callable function which is called
+                when the response generated by the model is invalid after
+                `max_retries` retries.
+        """
+        # step1: prepare keyword arguments
+        kwargs = {**self.generate_args, **kwargs}
+
+        # step2: forward to generate response
+        response = self.client.embeddings.create(
+            input=texts,
+            model=self.model_name,
+        )
+
+        # step3: record the model api invocation if needed
+        self._save_model_invocation(
+            arguments={
+                "model": self.model_name,
+                "input": texts,
+                **kwargs,
+            },
+            response=response.model_dump(),
+        )
+
+        # step4: update monitor accordingly
+        self.update_monitor(call_counter=1, **response.usage.model_dump())
+
+        # step5: return response
+        response_json = response.model_dump()
+        if len(response_json["data"]) == 0:
+            return ModelResponse(
+                embedding=response_json["data"]["embedding"][0],
+                raw=response_json,
+            )
+        else:
+            return ModelResponse(
+                embedding=[_["embedding"] for _ in response_json["data"]],
+                raw=response_json,
+            )
+
+    def _register_default_metrics(self) -> None:
+        # Set monitor accordingly
+        # TODO: set quota to the following metrics
+        self.monitor.register(
+            self._metric("call_counter"),
+            metric_unit="times",
+        )
+        self.monitor.register(
+            self._metric("prompt_tokens"),
+            metric_unit="token",
+        )
+        self.monitor.register(
```

Comment:
does embedding model response have "completion_tokens" information?

Reply:
They include a `"completion_tokens":0` in their examples, so I added this just to be save. (See https://open.bigmodel.cn/dev/api#text_embedding)